### DISCUSSION TITLE
How to get column value enumerator If I don't know ahead of time the types for each column

### DISCUSSION THREAD

**Discussion by zhengshy2016** [G-Research/ParquetSharp#394] [Q&A]
Hey, guy. I am trying to get the column value enumerator, but I don't know ahead of time the types of each column. Is this the correct way to get the column value enumerator? Thanks for the help.

```
var columnEnumerator = rowGroupReader.Column(0).LogicalReader().Apply(new ColumnValuesEnumerator());

sealed class ColumnValuesEnumerator : ILogicalColumnReaderVisitor<IEnumerator>
{
        public IEnumerator OnLogicalColumnReader<TElement>(LogicalColumnReader<TElement> logicalColumnReader)
        {         
            return logicalColumnReader.GetEnumerator();
        }
}
```

**Comment by adamreeve** [comment]
Hi @zhengshy2016 

Yes, although you're returning an untyped enumerator, the `IEnumerable` interface is only implemented by the typed `LogicalColumnReader<TElement>` class, so using an `ILogicalColumnReaderVisitor` as you've demonstrated is the best approach.
